### PR TITLE
Add batch queries

### PR DESCRIPTION
**Pull request by SpencerC**

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -112,3 +134,67 @@ func ExampleClient_Query_scan() {
...
+func ExampleClient_Batch() {
+	c, err := NewStaticTokenClient(
+		endpoint, token,
+		WithDefaultKeyspace("example"),
+	)
+	if err != nil {
+		log.Fatalf("failed to initialize client: %v", err)
+	}
+
+	err = c.Batch(
```
Comment (mpenick) [inline client_test.go:147]: This is a nice syntax for this. Then if you want you want to programmatically build a batch you would create a slice?

```go
batch := make([]*Querys, 0, 0)

// Logic

batch = append(batch, c.Query(`....`)

// ...

c.Batch(BatchUnlogged, batch...)
```

Comment (SpencerC) [reply client_test.go:147]: Correct

File: client.go [hunk 2]
```diff
@@ -131,6 +131,15 @@ func (c *Client) Query(cql string, values ...any) *Query {
 	}
 }
 
+// Batch creates a new Astra batch query.
+func (c *Client) Batch(batchType BatchType, queries ...*Query) *BatchQuery {
```
Comment (mpenick) [inline client.go:135]: I think it makes more sense to not include `BatchType` in the parameters. Internally, `.batchType` should default to `BatchLogged` and is most likely the correct choice for most cases, multi-partition mutations will be logged and single partition mutations will not be logged. 

Then, the `BatchQuery` type should provide a method `.BatchType(batchType BatchType)` like you did for `.Keyspace()` to override if a user decides they want to have unlogged multi-partition batches or if they're using counters.

```go
func (b *BatchQuery) BatchType(batchType BatchType) *BatchQuery {
	b.batchType = batchType
	return b
}
```

Comment (aar0np) [reply client.go:135]: I completely agree with defaulting to "LOGGED" unless specified differently.  And the use cases for "UNLOGGED" are (should be) small in number and infrequent.

**Review by aar0np — APPROVED** [review]

**Pull request merged**